pilot/pkg/bootstrap/server.go

```diff
@@ -1347,6 +1352,29 @@ func (s *Server) serveHTTP() error {
 	return nil
 }
 
+func (s *Server) initReadinessProbes() {
+	probes := map[string]readinessProbe{
+		"discovery": func() (bool, error) {
+			return s.XDSServer.IsServerReady(), nil
+		},
+		"sidecarInjector": func() (bool, error) {
+			if s.readinessFlags.sidecarInjectorReady {
+				return true, nil
+			}
+			return false, fmt.Errorf("sidecar injector not ready")
```

Comment:
On L1358 we return `ready, nil`. But here we return error. Should they be consistent?

Reply:
I think we can just make them return a `bool`, then we can use the `name` which we already have to make the error useful.